scripts/env_wrapper.sh

```diff
@@ -45,12 +59,49 @@ fi
 cmd=("$@")
 if should_use_lima; then
     setup_lima
-    cache_dir="/home/debian/mkosi-cache"
-    cache_cmd="mkdir -p \"$cache_dir\" || true"
+
+    mkosi_cache=/home/debian/mkosi-cache
+    mkosi_output=/home/debian/mkosi-output
+
     if [[ "${cmd[0]}" == "mkosi" ]]; then
-        cmd+=("--cache-directory=$cache_dir")
+        limactl shell "$LIMA_VM" mkdir -p "$mkosi_cache" "$mkosi_output"
+
+        cmd=(
+            # First off, we need to run mkosi in a new user namespace,
+            # because it creates files owned by multiple uids/gids, and we
+            # either need to run as root, or use this unshare, which is safer.
+            # See also:
+            # https://manpages.debian.org/unstable/mkosi/mkosi.1.en.html#:~:text=Why%20do%20I%20see%20failures%20to%20chown%20files%20when%20building%20images
+            "unshare"
+            "--map-auto" "--map-current-user"
+            "--setuid=0" "--setgid=0"
+            "--"
+            # Next, we use which because actual file is somewhere in /nix/store
+            # and it's easier to explicitly specify it here instead of passing
+            # $PATH.
+            '$(which mkosi)'
+            # Pass all original arguments except the first one (mkosi)
+            "${cmd[@]:1}"
+            # We can't use default cache dir from mnt/, because it is mounted
+            # from host, and mkosi will try to preserve root/other permissions
+            # without success.
+            "--cache-directory=$mkosi_cache"
+            # For the same reason, we need to use separate output dir.
+            # mkosi tries to preserve ownership of output files, which fails,
+            # as it is running from root in a user namespace.
+            "--output-dir=$mkosi_output"
+        )
     fi
-    limactl shell --workdir "/home/debian/mnt" "$LIMA_VM" bash -c "$cache_cmd; nix develop -c ${cmd[*]@Q}"
+
+    limactl shell "$LIMA_VM" bash -c \
+        "cd /home/debian/mnt && nix develop -c bash -c '${cmd[*]}'"
+
+    limactl shell "$LIMA_VM" mkdir -p /home/debian/mnt/build
+    # TODO: quoting & run only after mkosi commands
+    limactl shell "$LIMA_VM" bash -c "cp -rv $mkosi_output/* /home/debian/mnt/build/ || true"
```

Comment:
I think you can do \"$mkosi_output\"/* here? Not sure Also we might want to consider mving instead of cping since the files could be pretty big